login.js

```diff
@@ -0,0 +1,72 @@
+// 이메일 input 요소 가져오기
+let emailInput = document.getElementById("email");
+let emailError = document.getElementById("email-error");
+
+// 비밀번호 input 요소 가져오기
+let passwordInput = document.getElementById("password");
+let passwordError = document.getElementById("password-error");
+
+// 비밀번호 토글 이미지 가져오기
+let passwordToggle = document.getElementById("password-toggle");
+
+let emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+
+// 비밀번호 가시성, 비가시성 토글 함수
+function togglePassword() {
+  if (passwordInput.type === "password") { // 비밀번호 보이게
+    passwordInput.type = "text";
+    passwordToggle.src = "assets/img/eye-on.svg";
+  } else { // 비밀번호 가리기
+    passwordInput.type = "password";
+    passwordToggle.src = "assets/img/eye-off.svg";
+  }
+}
+
+// 이메일 입력란 포커스를 잃었을 때 유효성 검사
+emailInput.addEventListener('focusout', function () {
+  if (emailInput.value === "") { // 이메일이 비어있을 때
+    emailError.textContent = "이메일을 입력해주세요.";
+    emailError.style.display = "block";
+    emailInput.style.border = "1px solid red";
+  } else if (!emailPattern.test(emailInput.value)) { // 올바른 이메일 형식이 아닐 때
+    emailError.textContent = "올바른 이메일 주소가 아닙니다.";
+    emailError.style.display = "block";
+    emailInput.style.border = "1px solid red";
+  } else { // 유효한 이메일 형식일 때
+    emailError.style.display = "none";
+    emailInput.style.border = "1px solid black";
+  }
+});
```

Comment:
addEventListener 후에 위치하는 콜백 함수는 인라인 익명 함수로 선언하기보다, 외부에 분리시켜서 작업해두는 게 좋을 것 같아요. 읽기 복잡해지고 함수 값을 자바스크립트가 기억하지 못하는데 오는 문제가 생기거든요!

```suggestion
function emailInputFocusOut() { ... }

emailInput.addEventListener('focusout', emailInputFocusOut);
```